File: plugins/ai.js
const config = require('../config');
const { cmd, commands } = require('../command');
const { fetchJson } = require('../lib/functions');
const sensitiveData = require('../dila_md_licence/a/b/c/d/dddamsbs');

cmd({
  pattern: "ai",
  desc: "AI chat",
  category: "main",
  react: "",
  filename: __filename
}, async (conn, mek, m, { from, quoted, body, isCmd, command, args, q, isGroup, sender, senderNumber, botNumber2, botNumber, pushname, isMe, isOwner, groupMetadata, groupName, participants, groupAdmins, isBotAdmins, isAdmins, reply }) => {
  try {
    // Define the new API URL with the new key
    const apiUrl = `https://generativelanguage.googleapis.com/v1beta/models/gemini-1.5-flash-latest:generateContent?key=AIzaSyByitRjKUtDonuVtpJm1R-RSdPdFPf-tcY`;
    
    // Prepare the body content for the API request
    const bodyContent = {
      contents: [{ parts: [{ text: q }] }]
    };
    
    // Fetch the AI response
    let data = await fetchJson(apiUrl, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify(bodyContent)
    });

    // Log specific parts of the response for debugging (optional)
    console.log('API response status:', data.status);
    console.log('API response error:', data.error); // Only log if an error exists

    // Check if the response structure is as expected
    if (data && data.contents && data.contents[0] && data.contents[0].parts && data.contents[0].parts[0].text) {
      let response = data.contents[0].parts[0].text;

      // Compose the message to send
      let replyText = `\n${sensitiveData.aiChatHeader}\n\n *𝗤𝘂𝗲𝗿𝘆*: _${q}_\n\n *𝗥𝗲𝘀𝗽𝗼𝗻𝘀𝗲*: _${response}_\n\n${sensitiveData.siteUrl}\n${sensitiveData.footerText}`;
      
      // Send the message with the AI response
      let sentMessage = await conn.sendMessage(from, { image: { url: sensitiveData.imageUrl }, caption: replyText }, { quoted: mek });
      
      // React to the sent message
      await conn.sendMessage(from, { react: { text: "✅", key: mek.key } });
      await conn.sendMessage(from, { react: { text: "", key: sentMessage.key } });
    } else {
      // Log the error for debugging
      console.error('Invalid response format:', data);
      reply(`Error: Invalid response format from AI API.`);
    }

  } catch (error) {
    console.log(error);
    
    // Check for specific error types from the API (if available)
    if (error.response && error.response.data && error.response.data.error) {
      reply(`Error: ${error.response.data.error.message || 'Unknown error occurred.'}`);
    } else {
      reply(`Error: ${error.message}`);
    }
  }
});

File: plugins/ai.py
import requests
import discord
from discord.ext import commands

# Replace with your bot token and command prefix
TOKEN = 'your-bot-token'
PREFIX = '!'

intents = discord.Intents.default()
intents.message_content = True
client = commands.Bot(command_prefix=PREFIX, intents=intents)

@client.event
async def on_ready():
    print(f'Logged in as {client.user}')

@client.command(name='ai', help='AI chat')
async def ai(ctx, *, query: str):
    try:
        # Fetch response from AI API
        response = requests.get(f'https://chatgptforprabath-md.vercel.app/api/gptv1?q={query}')
        data = response.json()
        reply_text = data.get('data', '')

        # Format the reply
        reply_message = (
            f"*𝗗𝗶𝗹𝗮𝗠𝗗 𝗔𝗜 𝗖𝗵𝗮𝘁 🧠*\n\n"
            f"🔍 *𝗤𝘂𝗲𝗿𝘆*: _{query}_\n\n"
            f"💬 *𝗥𝗲𝘀𝗽𝗼𝗻𝘀𝗲*: _{reply_text}_\n\n"
            f"dilalk.vercel.app\nᵐᵃᵈᵆ ᵇʸ ᵐʳᵈⁱˡᵃ ᵒᶠᶜ"
        )

        # Send the reply with the image
        image_url = 'https://telegra.ph/file/5aee066590cf0c6eabf9d.jpg'
        embed = discord.Embed(description=reply_message)
        embed.set_image(url=image_url)

        await ctx.send(embed=embed)

    except Exception as e:
        await ctx.send(f'Error: {str(e)}')

client.run(TOKEN)
